File: .NET API/Models/DTO/MealReviewDTO/GetMealReviewsRequest.cs
namespace FoodDelivery.Models.DTO.MealReviewDTO;

public record GetMealReviewsRequest
{
    public Guid MealID { get; init; }
    public float Rating { get; init; }
    public string MealName { get; init;}
    public int FiveStarCount { get; init; }
    public int FourStarCount { get; init; }
    public int ThreeStarCount { get; init; }
    public int TwoStarCount { get; init; }
    public int OneStarCount { get; init; }
    public ICollection<GetMealReview> MealReviews { get; init; }
}

public record GetMealReview
{
    public string CustomerName { get; init; }
    public string CustomerImage { get; init; }
    public string ReviewText { get; init; }
    public int Rating { get; init; }
    public DateTime ReviewDate { get; init; }
}

File: Flutter/features/meal/models/get_meal_reviews_request.dart
class GetMealReviewsRequest {
  String? customerName;
  String? customerImage;
  String? reviewText;
  int? rating;
  String? reviewDate;

  GetMealReviewsRequest(
      {this.customerName,
      this.customerImage,
      this.reviewText,
      this.rating,
      this.reviewDate});

  GetMealReviewsRequest.fromJson(Map<String, dynamic> json) {
    customerName = json['customerName'];
    customerImage = json['customerImage'];
    reviewText = json['reviewText'];
    rating = json['rating'];
    reviewDate = json['reviewDate'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['customerName'] = customerName;
    data['customerImage'] = customerImage;
    data['reviewText'] = reviewText;
    data['rating'] = rating;
    data['reviewDate'] = reviewDate;
    return data;
  }
}
